orchestrator/orchestrator.go

```diff
@@ -148,37 +165,15 @@ func (o *Orchestrator) Stop(ctx context.Context) error {
 }
 
 func (o *Orchestrator) kickOffMining(ctx context.Context) error {
-	var once sync.Once
-	var err error
-	ctx, cancel := context.WithCancel(ctx)
-
-	handleErr := func(e error) {
-		if e == nil {
-			return
-		}
-
-		once.Do(func() {
-			err = e
-			cancel()
-		})
+	if err := o.l1Chain.SetIntervalMining(ctx, nil, 2); err != nil {
+		return errors.New("failed to start interval mining on l1")
 	}
-
-	var wg sync.WaitGroup
-	wg.Add(len(o.l2Chains) + 1)
-
-	handleErr(o.l1Chain.SetIntervalMining(ctx, nil, 2))
-	wg.Done()
-
 	for _, chain := range o.l2Chains {
-		go func() {
-			defer wg.Done()
-			handleErr(chain.SetIntervalMining(ctx, nil, 2))
-		}()
+		if err := chain.SetIntervalMining(ctx, nil, 2); err != nil {
```

Comment:
previously when this was in a go routine we could kick off mining on each chain in parallel, this reduces clock drift and also speeds up the startup time. is there a reason you prefer this to be synchronous?

Reply:
ahh wait i'm seeing `go func() { .. }` within that for loop now. Let me revert that piece. So it's concurrent

Let me revert this

Reply:
okay reverted to make this concurrent, good catch. Missed that